From a17aa2b22ee941ea9c2953bbef0efee35024da6c Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert@linux-m68k.org>
Date: Tue, 18 May 2021 13:57:54 +0200
Subject: [PATCH 54/61] video: fbdev: starfive: Fix frame buffer reserved
 memory resource conflict

    starfive,vpp-lcdc 12000000.sfivefb: can't request region for resource [mem 0xfb000000-0xfcffffff]
    starfive,vpp-lcdc 12000000.sfivefb: Fail to allocate video RAM
    starfive,vpp-lcdc 12000000.sfivefb: starfive fb init fail
    starfive,vpp-lcdc 12000000.sfivefb: fb info init FAIL
    starfive,vpp-lcdc: probe of 12000000.sfivefb failed with error -16

devm_ioremap_resource() calls devm_request_mem_region(), which fails as
the reserved memory for the frame buffer is already present in the
resource list, cfr. /proc/iomem:

    fb000000-fcffffff : Reserved

Fix this by mapping the frame buffer memory using devm_ioremap().

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
---
 drivers/video/fbdev/starfive/starfive_fb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/starfive/starfive_fb.c b/drivers/video/fbdev/starfive/starfive_fb.c
index 865471f06609..73bad88a1029 100644
--- a/drivers/video/fbdev/starfive/starfive_fb.c
+++ b/drivers/video/fbdev/starfive/starfive_fb.c
@@ -656,7 +656,8 @@ static int sf_fb_map_video_memory(struct sf_fb_data *sf_dev)
 	sf_dev->fb.screen_size = resource_size(&res_mem);
 	sf_dev->fb.fix.smem_start = res_mem.start;
 
-	sf_dev->fb.screen_base = devm_ioremap_resource(sf_dev->dev, &res_mem);
+	sf_dev->fb.screen_base = devm_ioremap(sf_dev->dev, res_mem.start,
+					      resource_size(&res_mem));
 	if (IS_ERR(sf_dev->fb.screen_base))
 		return PTR_ERR(sf_dev->fb.screen_base);
 
-- 
2.30.2

